### ISSUE TITLE
Enum types use enum value rather than enum name when converted to string

### ISSUE THREAD

**Issue by synergiance** [vrchat-community/UdonSharp#71]
**Describe the bug in detail:**
In C#, when you use ToString() on an enum type, it will return the name of the enum value, while in U# it returns the integer converted to a string. This is undesirable behavior most of the time, and unaligned with C#'s behavior.

**Provide steps/code to reproduce the bug:**
Let's take this enum for instance:
```cs
public enum MyEnum {
    MyFirstValue,
    MySecondValue
}
```

If we print it out like this, we should see the problem:
```cs
MyEnum myEnumInstance = MyEnum.MyFirstValue;
Debug.Log(myEnumInstance.ToString());
```

**Expected behavior:**
Log should read "MyFirstValue"

**Observed behavior:**
Log actually reads "0"

**Comment by MerlinVR** [comment]
This is known and will have partial handling eventually but can't have full handling due to boxing obscuring the user type.

**Comment by jellejurre** [comment]
Maybe add it to the "Differences from regular Unity C# to note" list at https://udonsharp.docs.vrchat.com/?

**Comment by synergiance** [comment]
Currently I work around it by creating a string array containing the names of the enum values.

Unfortunately though the workaround splits these into two separate parts of the code, which is undesirable.
Is it possible to just internally create this structure and any enum tostring just reference the name array?

**Comment by MerlinVR** [comment]
Yeah that's the plan to generate arrays, you can make a PR for it if you want it before I get around to it. If you do note that enums with the Flags attribute generate their ToString() value differently from ones without it.

**Comment by synergiance** [comment]
Speaking of enums using the flags attribute, it appears that operator | confuses the symbol processor on flags enums.

**Comment by MerlinVR** [comment]
Only equality and inequality are handled for enums at the moment
